### PR TITLE
fix(example): add dns resolve for http_proxy

### PR DESCRIPTION
**Pull request by kaiwk**
see #2103

### CODE REVIEW

File: examples/http_proxy.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #![deny(warnings)]
 
 use std::convert::Infallible;
-use std::net::SocketAddr;
+use std::net::{SocketAddr, ToSocketAddrs};
```
Comment (davidbarsky) [inline examples/http_proxy.rs]: Tokio now has an asynchronous DNS resolution function: https://docs.rs/tokio/0.2.8/tokio/net/fn.lookup_host.html. Maybe consider using that instead, as std's `to_socket_addrs()` is a blocking operation.

Comment (seanmonstar) [reply examples/http_proxy.rs]: Or another option, that doesn't require enabling another Tokio feature, is to just use `hyper::client::connect::dns::GaiResolver`.

Comment (kaiwk) [reply examples/http_proxy.rs]: > Tokio now has an asynchronous DNS resolution function: https://docs.rs/tokio/0.2.8/tokio/net/fn.lookup_host.html. Maybe consider using that instead, as std's `to_socket_addrs()` is a blocking operation.

it seems we need upgrade `tokio` to use `lookup_host`.

I  `pub` the  `resolve` method in `dns.rs`, is it appropriate?

**Comment by shirshak55** [comment]
is there any reason not to use to_socket_addr which comes with std library?
https://github.com/shirshak55/http-to-socks/blob/master/src/main.rs#L74

**Comment by shirshak55** [comment]
Is this pr going to be merged?

**Comment by seanmonstar** [comment]
I'd rather not make some of these functions part of the public API. The example includes in its comments what it can and cannot do, and even with DNS resolution, it won't work over HTTPS. I think for that reason it's better to show it in a limited capacity, and imply that making a full proxy is more involved.

**Pull request closed without merging**